### ISSUE TITLE
suggestion plndr-cloud-provider pod should be deployed as a Deployment or StatefulSet

### ISSUE THREAD

**Issue by cameronbraid** [plunder-app/kube-vip#50]
Currently if the plndr-cloud-provider pod is deleted (evicted etc.. or accidentally) it doesn't get re-created

The simplest solution is to move to a statefulset with replicas 1

Justification for StatefulSet is from https://github.com/tektoncd/pipeline/pull/2630

> Singleton workloads can be implemented in multiple ways, and they differ in behavior when the Node becomes unreachable:
> * as a Pod - the Pod is not managed, so it will not be recreated.
> * as a Deployment - the Pod will be recreated and puts Availability before the singleton property
> * as a StatefulSet - the Pod will be recreated but puds the singleton property before Availability

```
apiVersion: apps/v1
kind: StatefulSet
metadata:
  name: plndr-cloud-provider
spec:
  podManagementPolicy: OrderedReady
  replicas: 1
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      app: kube-vip
      component: plndr-cloud-provider
  template:
    metadata:
      labels:
        app: kube-vip
        component: plndr-cloud-provider
    spec:
      containers:
      - command:
        - /plndr-cloud-provider
        image: plndr/plndr-cloud-provider:0.1.3
        name: plndr-cloud-provider
        imagePullPolicy: Always
        resources: {}
      dnsPolicy: ClusterFirst
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      terminationGracePeriodSeconds: 30
      serviceAccountName: plunder-cloud-controller
```

**Comment by thebsdbox** [comment]
Feel free to create a PR for this change to the manifest (or I can) :)

**Comment by thebsdbox** [comment]
Fixed in #100
